src/Umbraco.Web/Security/MembershipHelper.cs

```diff
@@ -289,11 +289,46 @@ public virtual IPublishedContent GetByProviderKey(object key)
             return MemberCache.GetByProviderKey(key);
         }
 
+        public virtual IEnumerable<IPublishedContent> GetByProviderKeys(IEnumerable<object> keys)
```

Comment:
For this class (since it may turn into an interface one day) we should only keep the IEnumerable methods and not have the `params` methods. Sorry i should have mentioned this previously.

Having both IEnumerable and params overloads is confusing but i understand can add value for API users but as it turns out we have an issue with UmbracoHelper and the IEnumerable<object> and the params object[] overloads since it isn't validating types. For example, it's perfectly legal to do this:

`helper.Content(new MyObject[] {null});`

This will use the `params object[]`  overload which means it is going to pass in an array of MyObject as the first item in the `object[]` array. 

... but that's another issue to be fixed ;)

Reply:
👍 params overloads are removed in 4d0d17e